### ISSUE TITLE
Extension is not working correctly 

### ISSUE THREAD

**Issue by MW-J97** [beltranaceves/random_picker#1]
Hi there,
Thanks for this great extension. 
But, I have an issue with it.
It's not responsive when the table is fully opened as a page. 
And only choose one row at the same time when the table is not opened as a page, I have to deselect the row to select another one.

**Comment by beltranaceves** [comment]
Hi, thanks. I'll take a look

**Comment by beltranaceves** [comment]
Got it working on my machine. I submitted the new version for review, it'll be published as soon as they approve it.

Notion made some changes and broke the script:
- Full-page collections have a different nested structure now, just had to change the selector
- The hover bar for selected items was wide but invisible, so it was covering the New and Random buttons. Injected some css to correct it

**Comment by MW-J97** [comment]
> Hi, thanks. I'll take a look

Okay, not it responds better. 
But, still has 2 downsides:
First, it chooses only one or at maximum 2 at once. And this is acceptable. But, before, it was selecting multiple items. 
Second, it only selects from the items seen at the current view. I have a very long table, it only chooses from the first items which I see now, not the below or above.

**Comment by beltranaceves** [comment]
- Selection of multiple elements was not intended. Since you want it I'll make it configurable, at least for table views so that users can choose
- Selection happens at the UI level, so until Notion opens up programmatic access, the extension can only select already loaded rows. I'm afraid you'll have to click on Load more at the bottom of the page before selecting random elements. 

you could check out [notion-enhancer](https://notion-enhancer.github.io/), it may have a plugin for autoloading databases

**Comment by beltranaceves** [comment]
I'm gonna check if I can implement the table autoloading myself

**Comment by MW-J97** [comment]
>  I'm afraid you'll have to click on Load more at the bottom of the page before selecting random elements.

I open the table as a page, so it should select of all items. 
Does load more is different than that?

**Comment by beltranaceves** [comment]
No, you are right. A table opened as a page should show all items right away

**Comment by MW-J97** [comment]
> No, you are right. A table opened as a page should show all items right away

Unfortunately, it doesn't select from all items. It selects only from the rows seen at the part of screen shown.

**Comment by MW-J97** [comment]
> No, you are right. A table opened as a page should show all items right away

Is there any updates for the selection in all items not the shown only?

**Comment by beltranaceves** [comment]
Not yet, I didn't have time to work on it this week

**Comment by MW-J97** [comment]
> Not yet, I didn't have time to work on it this week

Okay, take your time and thanks for this great effort. 
I just want to be sure that the issue is not for me only. It's doesn't select items from all the table when it's open as a page. Is that only for me? Or an issue?

**Comment by beltranaceves** [comment]
It's an issue. Full page tables load rows dynamically to improve performance, so at any given time it only shows n rows.

 The problem with that is that the extension pick from rows present in the UI, so unless I come up with another method it won't be able to select rows from non-visible parts of the table

**Comment by MW-J97** [comment]
> It's an issue. Full page tables load rows dynamically to improve performance, so at any given time it only shows n rows.
> 
> The problem with that is that the extension pick from rows present in the UI, so unless I come up with another method it won't be able to select rows from non-visible parts of the table

Okay, I got it now. Waiting for it. 
Thanks for your effort.

**Comment by MW-J97** [comment]
> It's an issue. Full page tables load rows dynamically to improve performance, so at any given time it only shows n rows.
> 
>  The problem with that is that the extension pick from rows present in the UI, so unless I come up with another method it won't be able to select rows from non-visible parts of the table

I also noted that multiple rows selection is not happening. It selects only one item at once. 

Another thing, can it change the UI of the seen rows to the area that has the selected row? For example, I now see the upper part of the table and the selected item is below in the table, so it will change the UI view to below of the table.

**Comment by MW-J97** [comment]
> It's an issue. Full page tables load rows dynamically to improve performance, so at any given time it only shows n rows.
> 
>  The problem with that is that the extension pick from rows present in the UI, so unless I come up with another method it won't be able to select rows from non-visible parts of the table


Hi there, 
Is there anything new on this issue?

**Comment by beltranaceves** [comment]
Sorry for taking so long. There have been a bunch of breaking changes from Notion. It won't work correctly until I port it to use the API

**Comment by MW-J97** [comment]
> Sorry for taking so long. There have been a bunch of breaking changes from Notion. It won't work correctly until I port it to use the API

No, doesn't matter. Thanks for your great effort.
So, it's applicable??

**Comment by MW-J97** [comment]
> Sorry for taking so long. There have been a bunch of breaking changes from Notion. It won't work correctly until I port it to use the API

@beltranaceves Hi there, is there anything new in this issue?

**Comment by MW-J97** [comment]
> Sorry for taking so long. There have been a bunch of breaking changes from Notion. It won't work correctly until I port it to use the API

@beltranaceves Hello, sorry to bother you. But, is there anything can be done for this issue, please? I use this extension a lot.

**Comment by beltranaceves** [comment]
I think you are the only user haha. I'm on vacation right now, I'll give this a try after the first week of may

**Comment by MW-J97** [comment]
> I think you are the only user haha. I'm on vacation right now, I'll give this a try after the first week of may

I think so, too, haha. Sorry for interrupting your vacation.
Thanks for help and waiting for you. It's really a great extension, and the only one with randomization option.

**Comment by beltranaceves** [comment]
Couldn't make it work. I'll wait until notion-enhancer finishes migration to manifest v3 and make their database autoloader available
